spark/dl/src/test/scala/com/intel/analytics/bigdl/optim/DistriOptimizerSpec.scala

```diff
@@ -136,6 +133,52 @@ class DistriOptimizerSpec extends FlatSpec with Matchers with BeforeAndAfter {
     }
   }
 
+  "DistriOptimizer" should "train all minibatches per epoch" in {
+    val numSamples = 64
+    val numClasses = 3
+    val height = 32
+    val width = 32
+    val images = Array.tabulate(64) { i =>
```

Comment:
It does. You can verify it by changing the line `if (recordsProcessedThisEpoch >= numSamples)` in DistriOptimizer to `if (recordsProcessedThisEpoch >= dataset.size())` as it was before, and the test will fail. That's also why I added the lines

````scala
assert(sampleDataSet.size() == numSamples)
assert(batchDataSet.size() == numBatches)
````

To make it clear that dataset.size() is different for each.